iron-dropdown.html

```diff
@@ -18,20 +61,198 @@
       Polymer({
         is: 'iron-dropdown',
 
+        behaviors: [
+          Polymer.IronControlState,
+          Polymer.IronA11yKeysBehavior,
+          Polymer.IronOverlayBehavior,
+          Polymer.NeonAnimationRunnerBehavior
+        ],
+
         properties: {
+          /**
+           * The orientation against which to align the dropdown content
+           * horizontally relative to the dropdown trigger.
+           */
+          horizontalAlign: {
+            type: String,
+            value: 'left'
+          },
 
           /**
-           * True if the menu is open.
+           * The orientation against which to align the dropdown content
+           * vertically relative to the dropdown trigger.
            */
-          opened: {
-            type: Boolean,
-            value: false,
-            reflectToAttribute: true
+          verticalAlign: {
+            type: String,
+            value: 'bottom'
+          },
+
+          /**
+           * The element that should be used to position the dropdown when
+           * it is opened.
+           */
+          positionTarget: {
+            type: Object,
+            observer: '_positionTargetChanged'
+          },
+
+          /**
+           * An animation config. If provided, this will be used to animate the
+           * opening of the dropdown.
+           */
+          openAnimationConfig: {
+            type: Object
+          },
+
+          /**
+           * An animation config. If provided, this will be used to animate the
+           * closing of the dropdown>
+           */
+          closeAnimationConfig: {
+            type: Object
+          },
+
+          _positionRectMemo: {
+            type: Object
+          }
+        },
+
+        listeners: {
+          'neon-animation-finish': '_onNeonAnimationFinish'
+        },
+
+        observers: [
+          '_updateOverlayPosition(verticalAlign, horizontalAlign)'
+        ],
+
+        attached: function() {
+          if (this.positionTarget === undefined) {
+            this.positionTarget = this.parentNode;
+          }
+        },
+
+        get containedElement() {
+          return Polymer.dom(
+            Polymer.dom(this.root).querySelector('content')
+          ).getDistributedNodes()[0];
+        },
+
+        get _positionRect() {
+          if (!this._positionRectMemo && this.positionTarget) {
+            this._positionRectMemo = this.positionTarget.getBoundingClientRect();
+          }
+
+          return this._positionRectMemo;
+        },
+
+        get _horizontalAlignTargetValue() {
+          if (this.horizontalAlign === 'right') {
+            return this._positionRect.right;
```

Comment:
oh man, since these are named perfectly, does `this._positionRect[this.horizontalAlign]` work? that would be magical